src/domain/community/community/community.entity.ts

```diff
@@ -0,0 +1,91 @@
+import { ID, Field, ObjectType } from '@nestjs/graphql';
+import {
+  BaseEntity,
+  Column,
+  Entity,
+  ManyToOne,
+  OneToMany,
+  OneToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { IGroupable } from '@src/common/interfaces/groupable.interface';
+import { UserGroup } from '@domain/community/user-group/user-group.entity';
+import { ICommunity } from './community.interface';
+import { Application } from '@domain/community/application/application.entity';
+import { Challenge, Ecoverse, Opportunity } from '@domain/challenge';
+
+@Entity()
+@ObjectType()
+export class Community extends BaseEntity implements ICommunity, IGroupable {
+  @Field(() => ID)
+  @PrimaryGeneratedColumn()
+  id!: number;
+
+  @Field(() => String, {
+    nullable: false,
+    description: 'The name of the Community',
+  })
+  @Column()
+  name: string;
+
+  @Field(() => String, {
+    nullable: false,
+    description: 'The type of the Community',
+  })
+  @Column()
+  type: string;
+
+  @OneToMany(
+    () => UserGroup,
+    userGroup => userGroup.community,
+    { eager: true, cascade: true }
+  )
+  groups?: UserGroup[];
+
+  @OneToMany(
+    () => Application,
+    application => application.community,
+    { eager: true, cascade: true }
+  )
+  applications?: Application[];
+
+  @OneToOne(
+    () => Ecoverse,
+    ecoverse => ecoverse.community,
+    { eager: false, cascade: false, onDelete: 'CASCADE' }
+  )
+  ecoverse?: Ecoverse;
+
+  @OneToOne(
+    () => Challenge,
+    challenge => challenge.community,
+    { eager: false, cascade: false, onDelete: 'CASCADE' }
+  )
+  challenge?: Challenge;
+
+  @OneToOne(
+    () => Opportunity,
+    opportunity => opportunity.community,
+    { eager: false, cascade: false, onDelete: 'CASCADE' }
+  )
+  opportunity?: Opportunity;
```

Comment:
onDelete: 'CASCADE' must be only on the one side of a relation otherwise will lead to cyclic reference.

Reply:
Fixed in https://github.com/cherrytwist/server/pull/821/commits/689c369cafab7d4253aa0b358830932b593ed16c